src/pip/_internal/commands/debug.py

```diff
@@ -134,10 +134,9 @@ def show_tags(options: Values) -> None:
 
 
 def ca_bundle_info(config: Configuration) -> str:
-    levels = set()
-    for key, _ in config.items():
-        levels.add(key.split(".")[0])
-
+    # Ruff misidentifies config as a dict.
+    # Configuration does not have support the mapping interface.
+    levels = {key.split(".")[0] for key, _ in config.items()}  # noqa: PERF102
```

Comment:
```suggestion
    levels = {key.split(".", 1)[0] for key, _ in config.items()}  # noqa: PERF102
```

While we’re at it, this should _actually_ improve the performance somewhat